src/silicone/database_crunchers/quantile_rolling_windows.py

```diff
@@ -216,7 +230,7 @@ def same_x_val_workaround(
                     weights /= sum(weights)
                     # We want to calculate the weights at the midpoint of step
                     # corresponding to the y-value.
-                    cumsum_weights = np.cumsum(weights)
+                    cumsum_weights = np.cumsum(weights) - 0.5 * weights
```

Comment:
The cumulative sum is no longer 1 right? But that's intended so that you get the right return value at the extremes?

Reply:
Correct. The cumulative sum is now the value at the midpoints, so we get the extreme value above half the weight of the largest element. Otherwise the weight of the largest element doesn't matter as much as the smallest element.